frontend/src/components/modals/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { useState, useRef } from 'react';
+import Alert from '@mui/material/Alert';
+import { useDispatch } from 'react-redux';
+import { Trans, useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import { ModalTypes, Modal } from './types';
+import { ModalContent } from './modalparts/ModalContent';
+import { ModalFooter } from './modalparts/ModalFooter';
+import { ModalFooterButtonDiv } from './modalparts/ModalFooterButtonDiv';
+import { ModalHeader } from './modalparts/ModalHeader';
+import { errorState, Input } from '../forms/FormField';
+import '../../shared.scss';
+import css from './ChangePasswordModal.module.scss';
+
+import { LoadingSpinner } from '../LoadingSpinner';
+import { StoreDispatchType } from '../../redux';
+import { userActions } from '../../redux/user';
+
+const classes = classNames.bind(css);
+
+export const ChangePasswordModal: Modal<ModalTypes.CHANGE_PASSWORD_MODAL> = ({
+  closeModal,
+  id,
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch<StoreDispatchType>();
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const confirmationError = errorState(useState(''));
+
+  const errorHandler = (response?: { data: string; status: number }) => {
+    switch (response?.status) {
+      case 400: {
+        setErrorMessage(t('Bad request - password'));
+        break;
+      }
+      default:
+        setErrorMessage(t('Internal server error'));
+        break;
+    }
+  };
+
+  const handleConfirm = async () => {
+    if (!formRef.current?.checkValidity()) return;
```

Comment:
Using the submit functionality on the form might be a better for this. This also allows submitting by pressing <kbd>enter</kbd> like other forms. The form will take care of the validation step, but the default action (posting the form) needs to be prevented.

```suggestion
  const handleConfirm = async (e: SyntheticEvent) => {
    e.stopPropagation();
    e.preventDefault();
```

I guess this was pretty much copied from `ConfirmPasswordModal`, so it would be good to change there as well.

form submit 1/3

Reply:
Changed both ChangePassword and ConfirmPassword modals to use submit.